File: src/main/java/com/agentstore/common/exception/server/ServerException.java
package com.agentstore.common.exception.server;

import com.agentstore.common.exception.BusinessException;
import com.agentstore.common.exception.constants.ErrorCode;

public abstract class ServerException extends BusinessException {
    protected ServerException(ErrorCode errorCode, Object... messageArguments) {
        super(errorCode, messageArguments);
    }

    protected ServerException(ErrorCode errorCode, Throwable cause, Object... messageArguments) {
        super(errorCode, cause, messageArguments);
    }
}

File: src/main/kotlin/com/agentstore/common/exception/server/ServerException.kt
package com.agentstore.common.exception.server

import com.agentstore.common.exception.BusinessException
import com.agentstore.common.exception.constants.ErrorCode

abstract class ServerException : BusinessException {
    protected constructor(errorCode: ErrorCode, vararg messageArguments: Any?) : super(
        errorCode,
        *messageArguments,
    )

    protected constructor(
        errorCode: ErrorCode,
        cause: Throwable,
        vararg messageArguments: Any?,
    ) : super(errorCode, cause, *messageArguments)
}
